src/Monolog/ContaoTableHandler.php

```diff
@@ -0,0 +1,195 @@
+
+/**
+ * ContaoLogHandler
+ *
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+class ContaoTableHandler extends AbstractProcessingHandler
+{
+    use ContainerAwareTrait;
+
+    /**
+     * @var string
+     */
+    private $dbalServiceName = 'doctrine.dbal.default_connection';
+
+    /**
+     * @var Statement
+     */
+    private $statement;
+
+    /**
+     * Gets the service name for the DBAL database connection.
+     *
+     * @return string
+     */
+    public function getDbalServiceName()
+    {
+        return $this->dbalServiceName;
+    }
+
+    /**
+     * Sets the service name for the DBAL database connection.
+     *
+     * @param string $name
+     */
+    public function setDbalServiceName($name)
+    {
+        $this->dbalServiceName = $name;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function handle(array $record)
+    {
+        if (!$this->isHandling($record)) {
+            return false;
+        }
+
+        $record = $this->processRecord($record);
+
+        $record['formatted'] = $this->getFormatter()->format($record);
+
+        if (!isset($record['extra']['contao']) || !$record['extra']['contao'] instanceof ContaoContext) {
+            return false;
+        }
+
+        if (!$this->canWriteToDb()) {
+            return false;
+        }
+
+        try {
+            $this->write($record);
+        } catch (DBALException $e) {
+            return false;
+        }
+
+        $this->executeHook($record['message'], $record['extra']['contao']);
+
+        return false === $this->bubble;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @throws DBALException
+     */
+    protected function write(array $record)
+    {
+        /** @var \DateTime $date */
+        $date = $record['datetime'];
+
+        /** @var ContaoContext $context */
+        $context = $record['extra']['contao'];
+
+        $this->statement->execute(
+            [
+                'tstamp'   => $date->format('U'),
+                'text'     => specialchars((string) $record['formatted']),
+                'source'   => (string) $context->getSource(),
+                'action'   => (string) $context->getAction(),
+                'username' => (string) $context->getUsername(),
+                'func'     => (string) $context->getFunc(),
+                'ip'       => (string) $context->getIp(),
+                'browser'  => (string) $context->getBrowser(),
+            ]
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function getDefaultFormatter()
+    {
+        return new LineFormatter('%message%');
+    }
+
+    /**
+     * Verifies database connection and prepares the statement.
+     *
+     * @return bool
+     */
+    private function canWriteToDb()
+    {
+        if (null !== $this->statement) {
+            return true;
+        }
+
+        if (null === $this->container || !$this->container->has($this->dbalServiceName)) {
+            return false;
+        }
+
+        try {
```

Comment:
The try-catch block feels wrong in a method called "canWriteToDb". Why is this method needed in the first place? Could the statement not be prepared in the `write()` method, too, which is already wrapped in a try-catch block?